Add required_audience, required_issuer guide to jwt (#6665)
opensearch-project/documentation-website: _security/authentication-backends/jwt.md

`jwt_url_parameter` | If the token is not transmitted in the HTTP header but rather as an URL parameter, define the name of the parameter here.
`subject_key` | The key in the JSON payload that stores the username. If not set, the [subject](https://tools.ietf.org/html/rfc7519#section-4.1.2) registered claim is used.
`roles_key` | The key in the JSON payload that stores the user's roles. The value of this key must be a comma-separated list of roles.

> `required_audience` | A target audience of JWT stored in the JSON payload. This is a [aud claim of JWT](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3).

Reviewer: This is correct from a technical perspective. You may want to rephrase the first sentence for clarity though. For example "The name of the audience which the JWT must specify. This corresponds to the "aud" claim of the JWT"

> `required_audience` | A target audience of JWT stored in the JSON payload. This is a [aud claim of JWT](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3).

Reviewer 2: ```suggestion
`required_audience` | The name of the audience which the JWT must specify. This corresponds [`aud` claim of the JWT](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3).
```

> `required_issuer` | A target issuer of JWT stored in the JSON payload. This is a [iss claim of JWT](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1).

Reviewer 2: ```suggestion
`required_issuer` | The target issuer of JWT stored in the JSON payload. This corresponds to the [`iss` claim of the JWT](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1).
```
